### ISSUE TITLE
parse “corrections” in Nikon RAW data format V2.00

### ISSUE THREAD

**Issue by steko** [totalopenstation/totalopenstation#14]
The Nikon RAW data format V2.00 can have “corrections” like this one:

```
SS,55,1.700,6.718,143.6064,104.5304,15:39:06,P
SS,56,1.700,6.734,138.1288,104.4754,15:39:44,P
CO,HT changed at PT=56 Old HT=1.500m
CO,OLD=56 X-3.787 Y5.548 Z-0.543
CO,HT changed at PT=55 Old HT=1.500m
CO,OLD=55 X-4.239 Y5.190 Z-0.548
```

It shouldn't be too difficult to parse these data rows. An idea would be to store points in a //dict// instead of a //list//, so they can be retrieved (and sorted) by PID (unique Point ID). In that way, changing a point coordinate would just be a matter of

```
#!python

points[pid]['ht'] = new_value
```

I'll see if I can provide a patch.

---
- Bitbucket: https://bitbucket.org/steko/totalopenstation/issue/14
- Originally reported by: Stefano Costa
- Originally created at: 2010-02-09T09:37:32.845

**Comment by steko** [comment]
_From Stefano Costa on 2011-12-02 18:12:25+00:00_

See also <<issue 1>> about having a dictionary like interface to the processed data.

**Comment by steko** [comment]
_From Stefano Costa on 2014-02-16 15:41:41+00:00_

[formats] Parse corrections in Nikon RAW V2.00, closes #14

→ <<cset 7ca4b0f563cd0fad580b9df7b3daff3d47bf1129>>

**Comment by steko** [comment]
_From Stefano Costa on 2014-02-16 15:46:52+00:00_

The changes committed earlier are based on a wrong interpretation of the correction format.

**Comment by steko** [comment]
_From Stefano Costa on 2014-02-16 17:58:04+00:00_

Backout 30766de

To my best understanding, the “corrections” are already included in the
proper data and the comment lines are there just for reference, not for
being parsed. I may come back on this in the future but at this time I
think we should stop here until new data is available.

Closes #14

→ <<cset None>>
